### PR TITLE
add spi0 overlay to support sc16is752 on spi0

### PR DESCRIPTION
**Pull request by beikeland**

### CODE REVIEW

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/dts-v1/;
+/plugin/;
+
+/ {
+	compatible = "brcm,bcm2835", "brcm,bcm2708", "brcm,bcm2709";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: `    compatible = "brcm,bcm2835";` is enough.

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+	fragment@0 {
+		target = <&gpio>;
+		__overlay__ {
+			spi0_pins: spi0_pins {
+				brcm,pins = <10 9 11>; /* miso mosi sck was 19 20 21*/
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: The "was 19 20 21" will just confuse people.

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+				brcm,function = <4>; /* alt0? was 3 / alt4 */
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: `/* alt0 */` is enough of a comment.

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+			};
+
+			spi0_cs_pins: spi0_cs_pins {
+				brcm,pins = <8>;  /* ce0 was 18 */
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: `/* ce0 */`.

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+				sc16is752_clk: sc16is752_clk {
+					compatible = "fixed-clock";
+					#clock-cells = <0>;
+					clock-frequency = <14745600>;
+				};
+			};
+		};
+	};
+
+	fragment@2 {
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: You don't need this fragment for SPI0.

File: arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		target = <&aux>;
+		__overlay__ {
+			status = "okay";
+		};
+	};
+
+    __overrides__ {
+      int_pin = <&sc16is752>,"interrupts:0";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/sc16is752-spi0-overlay.dts]: Everything else is beautifully indented - use TABs here as well.

**Comment by pelwell** [comment]
By the way, there's no reason why this should be specific to the 1st generation Pis - even 40-pin header Pis can use SPI0 on GPIOs 7-11.

**Comment by pelwell** [comment]
You are also missing a fragment to disable spidev0.0:
```
        fragment@2 {
                target = <&spidev0>;
                __overlay__ {
                        status = "disabled";
                };
        };
```

**Comment by beikeland** [comment]
I *think* thats it sorted. I hardly know enough to be dangerous so fingers crossed..

**Comment by pelwell** [comment]
There are still some problems:
1. You've deleted the sc16is752 node, which makes the whole overlay a bit pointless.
2. You don't actually need fragment 0 or the pinctrl properties from fragment 1 - these are the defaults in the base device tree.

**Review by pelwell — COMMENTED** [review]
I'm reading it in a phone at the moment, but it looks OK. If you add the newline at the end again I'll do a quick final check when I'm back in front of a keyboard.

**Comment by pelwell** [comment]
Thanks - I added the documentation and Makefile entry, and merged it offline (https://github.com/raspberrypi/linux/commit/804d8f832ecde209afa6201c2e27aeb106e22396).

**Comment by beikeland** [comment]
Thanks for all your help and patience!

**Pull request closed without merging**